app/Commands/Dev/Core/Compile.hs

```diff
@@ -15,16 +16,26 @@ runCommand opts = do
   file <- getFile
   s <- embed (readFile (toFilePath file))
   tab <- getRight (mapLeft JuvixError (Core.runParserMain file Core.emptyInfoTable s))
-  C.MiniCResult {..} <- getRight (run (runError (coreToMiniC asmOpts tab :: Sem '[Error JuvixError] C.MiniCResult)))
-  buildDir <- askBuildDir
-  ensureDir buildDir
-  cFile <- inputCFile file
-  embed $ TIO.writeFile (toFilePath cFile) _resultCCode
-  Compile.runCommand opts {_compileInputFile = AppPath (Abs cFile) False}
+  case opts ^. compileTarget of
+    TargetGeb -> runGebPipeline opts file tab
+    _ -> runCPipeline opts file tab
   where
     getFile :: Sem r (Path Abs File)
     getFile = someBaseToAbs' (opts ^. compileInputFile . pathPath)
 
+runCPipeline ::
+  forall r.
+  (Members '[Embed IO, App] r) =>
+  CoreCompileOptions ->
+  Path Abs File ->
+  Core.InfoTable ->
+  Sem r ()
```

Comment:
Wrap the type arguments with a record type, say runCPipelineArg. We are using the convention in other places.

Reply:
I can do this, but seriously, here it's more trouble than it's worth. There are just 3 arguments to a simple local helper function. I don't think you're using this convention that consistently.

Reply:
Done

Reply:
I would say here was not needed, specially since all types are different